localstack_snapshot/snapshots/transformer.py

```diff
@@ -43,8 +47,20 @@ def new_scope(self, scope: str) -> int:
 def _register_serialized_reference_replacement(
     transform_context: TransformContext, *, reference_value: str, replacement: str
 ):
-    if '"' in reference_value:
-        reference_value = reference_value.replace('"', '\\"')
+    # Provide a better error message for the TypeError if the reference value is not iterable (e.g., float)
+    # Example: `TypeError: argument of type 'float' is not iterable`
+    # The snapshot library currently does not support reference replacements for non-string types.
+    try:
+        if '"' in reference_value:
+            reference_value = reference_value.replace('"', '\\"')
+    except TypeError as e:
+        message = (
+            f"Reference value {reference_value} of type {type(reference_value)} is not iterable."
+            f" Consider using `reference_replacement=False` in your transformer"
+            f" for the replacement {replacement}."
+        )
+        SNAPSHOT_LOGGER.error(message, exc_info=True)
+        raise TransformerException(message) from e
```

Comment:
I can imagine both options in this case.

I intuitively went for the Pythonic way of "Ask for forgiveness, not permission" and to make the confusing `TypeError` source more explicit. However, performance does not matter here and readability seems more important. Going for the string check also guards against other errors (e.g., @dfangl pointed out that using a list raises an `AttributeError` here because a list is iterable but fails at the `.replace` call). Hence, guarding against non-strings seems more appropriate here. Adjusted accordingly 👍